### ISSUE TITLE
Warning: file_get_contents(\/data\/config.json): failed to open stream: No such file or directory in...

### ISSUE THREAD

**Issue by ondrejhlavacek** [keboola/runner-sync-api#9]
Nějak se mi nedaří pustit žádnou sync akci. Tipuju, že to bude namountovaným /tmpem, přehodil jsem to z `/tmp/:/tmp/` (to mi hlásilo `chmod: changing permissions of '/tmp/': Operation not permitted`) na `./tmp/:/tmp/` (a zkusil jsem to i bez mountu)

**Comment by odinuv** [comment]
hm, me to jede, zkus si tam dat docker volume
./tmp nefunguje, protoze se mountuje jakoby dvakrat (jednou pri spusteni sync-api a pak vevnitr v docker run a tam se mu to splete)

**Comment by ondrejhlavacek** [comment]
nastavil jsem mount na `/tmp/runner-sync-api/:/tmp/` a dělá to furt.

**Comment by ondrejhlavacek** [comment]
a je potřeba ten /tmp vůbec mountovat? děje se to teda i když %tmp vůbec nemountuju :-(

**Comment by ondrejhlavacek** [comment]
když si pustím ty commandy v kontejneru ručně, tak to projde

**Comment by ondrejhlavacek** [comment]
ne! budu muset zkoumat dál :-(

**Comment by odinuv** [comment]
ta CF template ti udela ECS task, staci si ho spustit (ja to poustim na instanci docker sandbox testing, ktera ma potrebny prava), nenutim ti to, jen chci vyloucit, ze je to fakt nejaka chyba v kodu a ne jen problem s docker compose

**Comment by ondrejhlavacek** [comment]
a je to teda určitě kvůli tomu "dvojitýmu mountu", resp. mountu docker socketu. mount to bere z původní úrovně toho docker socketu, takže z mýho noťasu. inception. ECS vyzkouším, to podle mě bude šlapat.

**Comment by odinuv** [comment]
jo, me to taky blblo, a prave tam neslo kvuli tomu inception dat ten docker volume - uplne stejne to blbne i na docker-runner, ale takhle jak to je mi to chodi

**Comment by ondrejhlavacek** [comment]
ok, zkusím revertnout všechny změny v `docker-compose.yml` a fixnout oprávnění na `/tmp`

**Comment by ondrejhlavacek** [comment]
ne, ne a ne...

`chmod: changing permissions of '/tmp/': Operation not permitted`

jdu hledat, co to dělá, na můj systémovej `/tmp` si úplně šahat nechci nechat...

**Comment by ondrejhlavacek** [comment]
mám to

https://github.com/keboola/runner-sync-api/blob/919b4489c3516ad436eb1791325f253870dd53db/docker/start.sh#L5

je to tam nutný? nešlo by jen napsat do instrukcí, že má bejt writable?

**Comment by odinuv** [comment]
aha, no ja potrebuju udelat, aby byl writable pro apache process
ale nemusis to mit prece namapovany na `/tmp/:/tmp/`, `/mrkev/:/tmp/` by melo fungovat stejne, ne?

**Comment by ondrejhlavacek** [comment]
ten apache ale běží pod rootem, ne?

ten vnitřní docker mountuje zase `/tmp/*:/tmp/*`, takže `/mrkev/:/tmp/` nejede (to jsem zkoušel v různejch variantách asi 3x hnedka zkraje)

**Comment by odinuv** [comment]
ne, apache bezi pod wwwuser

**Comment by ondrejhlavacek** [comment]
hm, mám takovýhle oprávnění na `/tmp` a šlape to

```
$ ls -la /tmp
lrwxr-xr-x@ 1 root  wheel  11 Sep 25 11:11 /tmp -> private/tmp

$ls -la /tmp/
drwxrwxrwt  4 root   wheel  128 Oct  1 18:34 .
drwxr-xr-x  6 root   wheel  192 Sep 25 11:12 ..
drwxrwxrwx  2 ondra  wheel   64 Oct  1 16:00 docker
```
ten `docker` folder udělal runner.

**Comment by ondrejhlavacek** [comment]
to tam je default od systému, s tím jsem nic nedělal, `/tmp` by na linuxovejch strojích měl bejt čitelnej a zapisovatelnej všem, tj. defaultně 777. chmodovat ale může jen owner nebo root, proto mi ten `start.sh` failuje (běží pod mým userem)

**Comment by odinuv** [comment]
no to je divny, me to bez toho chmodu neslo, a jaktoze bezi start.sh pod tvym userem?

**Comment by ondrejhlavacek** [comment]
prdlajs, jsem neprobuzenej idiot, samozřejmě, že běží v kontejneru. tam běží pod tím `wwwuser`, že? pokud jo, tak ten taky chmodovat nemůže.

**Comment by odinuv** [comment]
v kontejneru bezi start.sh pod rootem

**Comment by ondrejhlavacek** [comment]
no, tak tam to stejně nejde :-(
(pouštím v kontejneru)
```
root@bdce2d373ba0:/code# ls -la /
total 68
drwxr-xr-x   1 root root 4096 Oct  2 06:50 .
drwxr-xr-x   1 root root 4096 Oct  2 06:50 ..
...
drwxrwxrwt   4 root root  128 Oct  2 06:50 tmp
...

root@bdce2d373ba0:/code# chmod a+rw /tmp  
chmod: changing permissions of '/tmp': Operation not permitted
```

**Comment by odinuv** [comment]
no dobre, ale bez toho chmodu ti to teda jede

**Comment by ondrejhlavacek** [comment]
jo, možná čas udělat zase nějakou výhybku pro windows? :-)

**Comment by odinuv** [comment]
no ten chmod je tam afaik potreba kvuli tomu ECS, na win me k nicemu neni
kazdopadne ted me napadlo - ono to mozna dela to `-R`
